editoast/src/views/path/path_item_cache.rs

```diff
@@ -51,17 +56,31 @@ impl PathItemCache {
             .map(|part| (infra_id, part.track.0.clone()))
             .chain(tracks);
         let existing_track_ids =
-            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks)
+            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks.clone())
                 .await?
                 .into_iter()
                 .map(|track| track.obj_id)
                 .collect();
+        let existing_track_labels =
+            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks)
+                .await?
+                .into_iter()
+                .filter_map(|track| {
+                    // Vérifie si track.extensions.sncf est Some, sinon retourne None
```

Comment:
I believe you can remove this comment since it describe exactly the code. If you really feel that's necessary, then translate it 😉